npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb

```diff
@@ -418,6 +429,24 @@ def fetch_lerna_packages
         dependency_files
       end
 
+      def fetch_pnpm_workspace_package_jsons
+        return [] unless parsed_pnpm_workspace_yaml["packages"]
+
+        package_json_files = []
+
+        workspace_paths(parsed_pnpm_workspace_yaml["packages"]).each do |path|
+          file = File.join(path, "package.json")
+
+          begin
+            package_json_files << fetch_file_from_host(file)
+          rescue Dependabot::DependencyFileNotFound
+            nil
```

Comment:
Is it expected this will happen sometimes?

Reply:
I think it is yeah.

Users may specify globs in their `pnpm-workspace.yml`, but it could happen that one of the folders that result from resolving the glob does not have a `package.json` file. That would be what this is rescuing. This is essentially copied from the logic of parsing the `workspaces` declaration in the `package.json` file which for `pnpm` lives in a separate file.

To be noted though, is that, we could probably use `Dir.glob` here directly (because we have the full repo cloned locally in this case). If we did that, there'd be no need for this rescue I think.

This was mentioned by @jakecoffman at https://github.com/dependabot/dependabot-core/pull/6932, and I suggested #6923 as a good moment to tackle it.

Reply:
If it's going away, then sounds fine... but if it's sticking around, explaining the expected context for why `nil` might be encountered is a useful code comment... that way if someone is debugging and hits nil, they can compare it to their scenario to see if it's an expected vs unexpected nil

Reply:
I added a comment a refactored this a bit at #7183.